### ISSUE TITLE
When reading's value is string type, it will be released prematurely

### ISSUE THREAD

**Issue by ych988** [edgexfoundry/device-sdk-c#493]
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 The issue is located in: 
data.c

### Is this a regression?
no

 Yes, the previous version in which this bug was not present was: ....

## 🔥 Exception or Error
<pre><code>

edgex_event_cooked *edgex_data_process_event
.....
      case IOT_DATA_STRING:
        iot_data_string_map_add (rmap, "value", iot_data_alloc_string (iot_data_string (values[i].value), IOT_DATA_REF));
        break;
.....

autoevent.c
 ae_runner →  edgex_data_process_event(...results...） → devsdk_commandresult_free (**results**, ai->resource->nreqs);

</code></pre>

The string of IOT_DATA_REF is member of "**results**" .  But，"**results**" will be release in devsdk_commandresult_free.
In other words, the processed event reading value using edgex_data_process_event  is released prematurely.
So that, reading value displays garbled code.

Replace IOT_DATA_REF with IOT_DATA_COPY to solve this problem,  

## 🌍  Your Environment
**Deployment Environment:**

**EdgeX Version [**REQUIRED**]:**
main

**Anything else relevant?**

**Comment by FelixTing** [comment]
fixed by #506
